### ISSUE TITLE
S2234 FN: Nullable argument value

### ISSUE THREAD

**Issue by H4pH** [SonarSource/sonar-dotnet#3879]
### Description

False negative with nullable argument value.

### Repro steps

This code doesn't throw by rule, but rule has been violated.
```csharp
void NotNullableParamVoid(int a, int b) 
{
    // Do nothing
}

void NullableParamValueVoid(int a, int? b)
{
    if (b.HasValue)
    {
        NotNullableParamVoid(b.Value, a);
    }
}
```

### Expected behavior

Rule understands this case.

### Related information

* C# 8.0
* dotnet 5.0.101
* SonarScanner for MSBuild 4.10
  Using the .NET Core version of the Scanner for MSBuild

**Comment by pavel-mikula-sonarsource** [comment]
Hi @H4pH ,

Thank you for reporting this case. I marked is a improvement of the rule as the FN case doesn't strictly violates the specification.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1931](https://sonarsource.atlassian.net/browse/NET-1931)

[NET-1931]: https://sonarsource.atlassian.net/browse/NET-1931?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
